### DISCUSSION TITLE
Deploying projects

### DISCUSSION THREAD

**Discussion by etyson17** [laravel-zero/laravel-zero#427] [Q&A]
Has anyone found a good way to deploy these projects? I was wondering if there was something like [goreleaser](https://goreleaser.com/) available for these types of projects. My end goal is to be able to push to gitlab and then be able to pull the compiled project from homebrew.

**Comment by Tsdjimmy** [comment]
Why not try writing a YAML file and using the repo actions

**Reply by etyson17** [reply]
We currently have a pipeline that we would like to replace and are looking for a more OOTB solution.

**Comment by owenvoke** [comment]
If you're wanting to push to Homebrew, you can try compiling a PHAR binary. This is built into Laravel Zero, and would involve basically just running `php <your-application> app:build`.

The resulting PHAR will be created in the `builds` directory, and you can push this to whatever storage you want (Git repo, GitLab storage of some sort, etc.).

Then, you can create a Brew formula that points to this PHAR (and depends on PHP).

I've written a pretty simple blog post which may help with this:
https://voke.dev/blog/distributing-laravel-zero-apps-with-brew
